util/chunk/compare.go

```diff
@@ -160,13 +166,19 @@ func cmpBit(l Row, lCol int, r Row, rCol int) int {
 	return lBit.Compare(rBit)
 }
 
-func cmpJSON(l Row, lCol int, r Row, rCol int) int {
-	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
-	if lNull || rNull {
-		return cmpNull(lNull, rNull)
+func getCmpJSONFunction(wp WarningReporter) CompareFunc {
+	// TODO: deduplicate the warning in a wider range
```

Comment:
We could compare against all warnings inside the sctx to check whether this warning is duplicated, but I'm not sure whether it's necessary.